projects/portal/src/app/app.component.ts

```diff
@@ -216,7 +216,8 @@ export class AppComponent implements OnInit {
     } else if (searchResult.type === 'txHash') {
       await this.router.navigate(['txs', searchResult.searchValue]);
     } else if (searchResult.type === 'block') {
-      await this.router.navigate(['explorer/blocks', searchResult.searchValue]);
+      const redirectUrl = `${location.protocol}//${location.host}/explorer/blocks/${searchResult.searchValue}`;
+      window.location.href = redirectUrl;
```

Comment:
I use `location.host` (with port No.) for debug in localhost:5000. 
But other part using `location.hostname`(without port No.) or using 'if' as below. ( staking.application.service.ts )

```
    const redirectUrl =
      location.port === '80' || location.port === '443' || location.port === ''
        ? `${location.protocol}//${location.hostname}/explorer/validators/${createValidatorData.validator_address}`
        : `${location.protocol}//${location.host}/explorer/validators/${createValidatorData.validator_address}`;
    window.location.href = redirectUrl;
```
I think both is correct , because /portal and /explorer is build on same port.